### DISCUSSION TITLE
Repeater inline editing

### DISCUSSION THREAD

**Discussion by pavelstianko** [directus/directus#17402] [Feature Requests]
It would be nice to have possibility to edit repeater directly on page (without side popup editing). If I want to edit multiple items, it is very slow and unproductive to edit all using popup. My idea is something like this:

https://user-images.githubusercontent.com/13132639/216766436-bdd0f5b5-b9b0-40ed-a1bd-66ced8f5d712.mov

**Comment by chris-zeitport** [comment]
💡 The repeater and the relation list (M2A, M2M) can be extended with an interface option where to open the details:
- drawer (side panel)
- inline (like in the video)
- focus, zen mode (replace current details, keep a header + breadcrumb)
- fullscreen (new tab)
- modal dialog (centered)

**Reply by pavelstianko** [reply]
Do you mean in native Directus or using custom extension?

**Reply by chris-zeitport** [reply]
My comment tries to extend your IDEA to a generic option native in the Directus app.
From an implementation point of view nearly all the building blocks are available. A custom extension would require to copy a lot of code from existing views. 

I am new to Directus, not a Contributor yet and had no time for a deep dive into the source code.

**Comment by LasseRosenow** [comment]
I love this design! It would improve directus A LOT from my POV :) ❤️

**Reply by pavelstianko** [reply]
Thank you! I think it can help a lot. The same functionality can be used for M2A, M2M and O2M. Something like "Collapse view" in Interface > Layout

**Comment by hancewin** [comment]
I didn't see this request before posting mine. I love the Design! This would be the perfect solution

**Comment by rijkvanzanten** [comment]
Closing this in favor of https://github.com/directus/directus/discussions/17792, as that's a more in-depth RFC on the same topic (I'll alter that one to apply to any relational interface instead of just the m2a 🙂 )
